algorithms/HPdclassifier/R/predict.hpdegbm.R

```diff
@@ -0,0 +1,218 @@
+#                                                                                   
+# This program is free software; you can redistribute it and/or                     
+# modify it under the terms of the GNU General Public License                       
+# as published by the Free Software Foundation; either version 2                    
+# of the License, or (at your option) any later version.                            
+#                                                                                   
+# This program is distributed in the hope that it will be useful,                   
+# but WITHOUT ANY WARRANTY; without even the implied warranty of                    
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                      
+# GNU General Public License for more details.                                      
+#                                                                                   
+# You should have received a copy of the GNU General Public License                 
+# along with this program; if not, write to the Free Software                       
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.    
+####################################################################################
+
+####################################################################################
+#  File hpdegbm.R
+#  Distributed version of gbm 
+
+#  This code is a distributed version of gbm function availabel in R gbm package.
+
+####################################################################################
+## A copy of gbm R package license:
+#
+#  Copyright (C) 1995-2012 The R Core Team
+#
+#  This program is free software; you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 2 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  A copy of the GNU General Public License is available at
+#  http://www.r-project.org/Licenses/
+####################################################################################
+
+
+####################################################################################
+###                     Data-distributed and model-centralized prediction function of distributed GBM
+####################################################################################
+predict.hpdegbm <- function(object, newdata, appType="binary-classification", type="link", trace = FALSE)
+{
+  # model: assembled GBM model
+  # best.iter: best iteration of sub-models
+  # newdata: testing data in dframe/darray
+  # appType: regression, binary-classification, multi-classification
```

Comment:
As far as I understand, appType is model-specific; i.e., it is not correct to use appType="binary-classification" when the model is a regression model or vice versa. In this case, the model type should automatically detected at the time of training and stored alongside the model. As an example, you can see that randomForest function also detects and returns the model type (regression, classification, and unsupervised).